### PR TITLE
Set all VPN pages to use subnav-refresh.html (Fixes #14712)

### LINKED ISSUE #14712: "Features" hyperlink disappears when accessing "Learn about VPNs" or "Download Mozilla VPN"
## Description

"Features" hyperlink disappears from the header sub-menu when accessing one of the other links

### Steps to reproduce

Be on https://www.mozilla.org/en-US/products/vpn/
Click on "Learn about VPNs" from the header sub-menu
Observe sub-menu behavior

https://github.com/mozilla/bedrock/assets/124033364/a26d2682-45f5-4f63-9564-12ff7d957d6d

### Expected result
All 3 hyperlinks are still present regardless of what page you've accessed

### Actual result

"Features" hyperlink disappears from the header sub-menu

### Environment

Issue reproduces for both staging and production environments

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

Removes the old subnav for VPN as it was leading to an inconsistent experience when browsing pages.

## Issue / Bugzilla link

#14712

## Testing

- [ ] http://localhost:8000/en-US/products/vpn/resource-center/
- [ ] http://localhost:8000/en-US/products/vpn/?xv=legacy
- [ ] http://localhost:8000/en-US/products/vpn/pricing/?xv=legacy

### CODE REVIEW

File: bedrock/products/templates/products/vpn/landing.html [hunk 1]
```diff
@@ -52,7 +52,7 @@
 {% endblock %}
 
 {% block sub_navigation %}
-  {% include 'products/vpn/includes/subnav.html' %}
+  {% include 'products/vpn/includes/subnav-refresh.html' %}
 {% endblock %}
```
Comment (janbrasna) [inline bedrock/products/templates/products/vpn/landing.html:56]: As it {% extends "products/vpn/base.html" %}, this block is already there:

https://github.com/mozilla/bedrock/blob/178aa6b762bff6b2741d5acd1e89f5d0e64ff1a7/bedrock/products/templates/products/vpn/base.html#L16-L18

right?

```suggestion
```

File: bedrock/products/templates/products/vpn/pricing.html [hunk 2]
```diff
@@ -37,7 +37,7 @@
 {% endblock %}
 
 {% block sub_navigation %}
-  {% include 'products/vpn/includes/subnav.html' %}
+  {% include 'products/vpn/includes/subnav-refresh.html' %}
 {% endblock %}
```
Comment (janbrasna) [inline bedrock/products/templates/products/vpn/pricing.html:41]: This one also {% extends "products/vpn/base.html" %}, so the same case?

https://github.com/mozilla/bedrock/blob/178aa6b762bff6b2741d5acd1e89f5d0e64ff1a7/bedrock/products/templates/products/vpn/base.html#L16-L18

```suggestion
```

**Review by janbrasna — CHANGES_REQUESTED** [review]
Thank you for this, it was super inconsistent once you also start switching languages (the og subnav had l10n logic on top of it).

I think now with only one nav, these are probably redundant?

**Comment by stephaniehobson** [comment]
Does this fix https://github.com/mozilla/bedrock/issues/13065 by any chance?

**Comment by janbrasna** [comment]
@stephaniehobson I understand the linked issue is mostly misinterpreting the role of subnav (and its utility of not changing around during navigation), and it's easy to understand why: the VPN subnav doesn't make use of the aria-current styling to denote the current page to help understand the local IA: 

https://github.com/mozilla/bedrock/blob/a81d4852e09a474e62b8be86f75316d38fb199f9/media/css/firefox/family/components/_subnav.scss#L139

Maybe if the included subnav gets migrated to the macro (setting these): https://github.com/mozilla/bedrock/blob/a81d4852e09a474e62b8be86f75316d38fb199f9/bedrock/base/templates/macros.html#L196

it might actually resolve said issue?

**Comment by alexgibson** [comment]
Thanks @janbrasna updated

@stephaniehobson I think https://github.com/mozilla/bedrock/issues/13065 got stuck because everyone has differing opinions about what should be in that nav. This was all pre-refresh though, so maybe we can just close that one out as wontfix?

**Review by craigcook — APPROVED** [review]
🥪

**Pull request merged**